evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -377,15 +377,17 @@ where
                 let ctx = MockCallDataContext {
                     calldata: call.input.clone(),
                     value: Some(call.transfer.value),
+                    reverts: false
                 };
                 if let Some(mock_retdata) = mocks.get(&ctx) {
                     return (Return::Return, Gas::new(call.gas_limit), mock_retdata.clone())
-                } else if let Some((_, mock_retdata)) = mocks.iter().find(|(mock, _)| {
+                } else if let Some((mock, mock_retdata)) = mocks.iter().find(|(mock, _)| {
                     mock.calldata.len() <= call.input.len() &&
                         *mock.calldata == call.input[..mock.calldata.len()] &&
                         mock.value.map(|value| value == call.transfer.value).unwrap_or(true)
                 }) {
-                    return (Return::Return, Gas::new(call.gas_limit), mock_retdata.clone())
+                    let return_type = if mock.reverts { Return::Revert } else { Return::Return };
+                    return (return_type, Gas::new(call.gas_limit), mock_retdata.clone())
```

Comment:
this returns the raw data without a selector prefix.

perhaps we should prepend this with the `ERROR_PREFIX`, see `encode_error`